api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```diff
@@ -0,0 +1,251 @@
+        // TODO: Mike is the correct and/or necessary?
+        Map<String,Object> mappings = new HashMap<String, Object>();
+        for (Parameter parameter : parameters) {
+            mappings.put(parameter.getName(), "${" + parameter.getName() + "}");
+        }
+        return mappings;
+    }
+
+    public static DataSetDefinition constructDataSetDefinition(DataSetDescriptor dataSetDescriptor, List<Parameter> parameters) {
+
+        DataSetDefinition dsd = null;
+
+        if ("sql".equalsIgnoreCase(dataSetDescriptor.getType())) {
+            dsd = constructSQLFileDataSetDefinition(dataSetDescriptor);
+        }
+        else {
+            log.error("Unsupported data set descriptor type: " + dataSetDescriptor.getType());
+        }
+
+        if (parameters != null) {
+            for (Parameter parameter : parameters) {
+                dsd.addParameter(parameter);
+            }
+        }
+
+        return dsd;
+    }
+
+    public static SqlFileDataSetDefinition constructSQLFileDataSetDefinition(DataSetDescriptor dataSetDescriptor) {
+
+        SqlFileDataSetDefinition dsd = new SqlFileDataSetDefinition();
+
+        File sqlFile = new File(getReportingDescriptorsConfigurationDir() + File.separator + dataSetDescriptor.getConfig());
+
+        if (sqlFile.exists()) {
+            dsd.setSqlFile(sqlFile.getAbsolutePath());
+        }
+        else {
+            log.error("SQL file " + dataSetDescriptor.getConfig() + " not found");
+        }
+
+        return dsd;
+    }
+
+    public static List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+
+        // currently only support a default, single CSV report design
+        ReportDesign csvDesign = constructCSVReportDesign(reportDefinition);
+
+        // TODO: more robust whitespace removal?  how do we want to determine the filename?
+        csvDesign.addPropertyValue(ReportDesignRenderer.FILENAME_BASE_PROPERTY, StringUtils.replace(reportDefinition.getName(), " ", ".").toLowerCase() + "." +
+                "{{ formatDate request.reportDefinition.parameterMappings.startDate \"yyyyMMdd\" }}." +
+                "{{ formatDate request.reportDefinition.parameterMappings.endDate \"yyyyMMdd\" }}." +
+                "{{ formatDate request.evaluateStartDatetime \"yyyyMMdd\" }}." +
+                "{{ formatDate request.evaluateStartDatetime \"HHmm\" }}");
+
+        return Collections.singletonList(csvDesign);
+    }
+
+    public static ReportDesign constructCSVReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign design = new ReportDesign();
+        design.setName("reporting.csv");
+        design.setReportDefinition(reportDefinition);
+        design.setRendererType(CsvReportRenderer.class);
+
+        // TODO: should we parameterize these? this was stuff I just copied from Mirebaials... are we okay with the default being ISO-8859-1; not sure what the blacklist does
```

Comment:
These seem very mirebalaisreports specific, would be good to better understand why they are there.  If they are so universal, then we should just bake them into the CsvReportRenderer implementation.  I think we should move these into the design config

Reply:
Yeah, I'm fine with moving them into a design config, will work on that... I don't know the history of these, for what it's worth, my hunch was they'd need to be parameterized, but figured I'd check first.